lighthouse-core/audits/byte-efficiency/offscreen-images.js

```diff
@@ -76,7 +76,8 @@ class OffscreenImages extends ByteEfficiencyAudit {
    */
   static computeWaste(image, viewportDimensions, networkRecords) {
     const networkRecord = networkRecords.find(record => record.url === image.src);
-    if (!image.resourceSize || !networkRecord) {
+    if (!image.resourceSize || !networkRecord || image.loading === 'lazy' ||
```

Comment:
Are the first two conditions here error cases (why don't we have resourceSize or the network record), and the latter two legitimate cases? I find the mix to be odd, perhaps a comment would clear it up.

Reply:
Not totally sure if the first two conditions are error cases. Seems like we'll always have `resourceSize` but it [could be 0](https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/gather/gatherers/image-elements.js#L208-L209), while the network record could be omitted for at least [this reason](https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/gather/gatherers/image-elements.js#L171-L172), maybe others? LMK if that's accurate, I could write a comment explaining these cases.

Reply:
First two cases are basically error cases, so I would agree with connor that we should separate and add a comment explaining. Something explaining why `loading="eager"` should pass would be great too.

```suggestion
    // If we don't know how big it was, we can't really report savings, treat it as passed.
    if (!image.resourceSize || !networkRecord) return null;
    // If the image had its loading behavior explicitly controlled already, treat it as passed.
    if (image.loading === 'lazy' || image.loading === 'eager') return null;
```